go/pkg/pwapi/api_challenge-subscription-validate.go

```diff
@@ -32,14 +34,44 @@ func (svc *service) ChallengeSubscriptionValidate(ctx context.Context, in *Chall
 		return nil, errcode.ErrGetChallengeSubscription.Wrap(err)
 	}
 
-	// FIXME: check if passphrase is valid
+	// check if challenge subscription is still open
+	if challengeSubscription.Status != pwdb.ChallengeSubscription_Active {
+		return nil, errcode.ErrChallengeValidationInactive.Wrap(err)
+	}
+
+	validPassphrases := []string{}
+	for _, instance := range challengeSubscription.SeasonChallenge.Instances {
+		var configData pwinit.InitConfig
+		err = json.Unmarshal(instance.GetInstanceConfig(), &configData)
+		if err != nil {
+			return nil, errcode.ErrParseInitConfig.Wrap(err)
+		}
+		matchCount := 0
+		tempPassphraseArray := []string{}
+		for _, userPassphrase := range in.Passphrases {
+			for _, challengePassphrase := range configData.Passphrases {
+				if userPassphrase == challengePassphrase {
+					tempPassphraseArray = append(tempPassphraseArray, userPassphrase)
+					matchCount++
+				}
+			}
+		}
+		if len(tempPassphraseArray) > len(validPassphrases) {
+			validPassphrases = tempPassphraseArray
+		}
+	}
```

Comment:
that's exactly the idea, I've noticed two things

1. you are storing in `validPassphrases` the passphrases hash, but they will change and have no values, what has value is the passphrase index, i.e. `[]int{0, 2, 3}` (that will be stored in JSON in a string at the end)
2. so you should refactor the algorithm to store what passphrases are valid vs missing (using a map will be a lot easier); and also count the amount input passphrases that don't match any instance
3. if you valid and invalid passphrases, I prefer to raise an error for the moment instead of just validating what was good, just raise an error that says that "some errors are good, but not all of them" instead of an opaque "invalid passphase(s)"

---

and small bonuses:

1. in go, we say slice instead of array (about the variable names)
2. matchCount seems to be uselss if I'm not wrong :)